.github/workflows/pr.yml

```diff
@@ -24,3 +23,40 @@ jobs:
         run: yarn test
       - name: yarn test:e2e
         run: yarn test:e2e
+  integration:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+      - name: Use Node.js 18
+        uses: actions/setup-node@v4
+        with:
+          node-version: "18"
+      - name: yarn install
+        run: yarn
+      - name: yarn start:docker
+        run: yarn start:docker
+      - name: yarn build
+        run: yarn build
+      # Borrowed from https://github.com/DCFoundation/cosmos-proposal-builder/blob/main/.github/workflows/pr.yml#L43-L61
+      - name: Wait for Docker container to be ready
+        run: |
+          timeout 180 bash -c '
+            TARGET_HEIGHT=1001
+            SLEEP=10
+            echo "Waiting for the Agoric service to be fully ready..."
+            echo "Target block height: $TARGET_HEIGHT"
+            while true; do
+              response=$(curl --silent http://localhost:26657/abci_info);
+              height=$(echo $response | jq -r ".result.response.last_block_height | tonumber");
+              if [ "$height" -ge $TARGET_HEIGHT ]; then
+                echo "Service is ready! Last block height: $height";
+                break;
+              else
+                echo "Waiting for last block height to reach $TARGET_HEIGHT. Current height: $height";
+              fi;
+              sleep $SLEEP;
+            done
+          '
+      - name: yarn start:contract
+        run: yarn start:contract
```

Comment:
Here's hoping we can test that the contracts actually got started.

One way is:

```
agoric follow -lF :published.agoricNames.instance | grep sellConcertTickets
```

but that presumes the `agoric` CLI is available; I'm not sure that's the case.

`agoric follow -lF ...` is just an HTTP GET, so it can be done with `curl` too.

I'm working on running the tests for contract functionality in an end-to-end context too. But we're not quite there yet.

Reply:
We do now! Jim's slides this morning gave me an idea and we're testing that. Though I don't know why the CI couldn't see `swaparoo` contract onchain. The other 2 contracts are available on based on the curl response.

Ref: https://github.com/Agoric/dapp-agoric-basics/actions/runs/8332287701/job/22801091735?pr=24

Reply:
> We do now!

nice!

> I don't know why the CI couldn't see swaparoo contract onchain.

very strange. it's certainly in the ci log: `[\\\\\\\"sellConcertTickets\\\\\\\",\\\\\\\"$23.Alleged: InstanceHandle\\\\\\\"]`

Reply:
It could grep `sellConcertTickets` and `postalService`, but not `swaparoo`

I tried to `CTRL+F` on the blob and I couldn't find `swaparoo` in the data blob. Do we need to wait for some time before curling and greping?

Reply:
oic. I was reading too quickly.

I'm not sure what's up with swaparoo there. But the way it's deployed changes significantly with the addition of governance (#21) so I'm not inclined to worry about it now.